File: PracticasPython/Practica4/FiguraGeometrica.py
from multipledispatch import dispatch
import math
@dispatch(float)
def calcular_area(radio):
    return math.pi * radio ** 2
@dispatch(float,float)
def calcular_area(base, altura):
    return base*altura
@dispatch(float, float, str)
def calcular_area(base, altura, tipo):
    if tipo == "triangulo":
        return (base*altura)/2
    return "tipo no valido"
@dispatch(float, float, float)
def calcular_area(base_mayor, base_menor, altura, tipo):
    if tipo == "trapecio":
        return ((base_mayor + base_menor)* altura) /2
    return "tipo no valido"
@dispatch(float, float, int)
def calcular_area(lado,apotema,lados):
    if lados == 5:
        perimetro = lado *lados
        return (perimetro * apotema) / 2
    return "no es un pentagono"
print("Área círculo:", calcular_area(5.0))
print("Área rectángulo:", calcular_area(4.0, 6.0))
print("Área triángulo rectángulo:", calcular_area(3.0, 5.0, "triángulo"))
print("Área trapecio:", calcular_area(8.0, 5.0, 4.0, "trapecio"))
print("Área pentágono:", calcular_area(6.0, 4.0, 5))

File: PracticasJava/Practica4/src/FiguraGeometrica.java

public class FiguraGeometrica {
	static double calcularArea(double radio) {
		return Math.PI * Math.pow(radio, 2);
	}
	static double calcularArea(double base, double altura) {
		return base*altura;
	}
	static double calcularArea(double base, double altura, boolean esTriangulo) {
		if(esTriangulo) {
			return (base*altura)/2;
		}
		return -1;
	}
	static double calcularArea(double baseMayor, double baseMenor, double altura, boolean esTrapecio) {
		if(esTrapecio) {
			return ((baseMayor+baseMenor)*altura)/2;
		}
		return -1;
	}
	static double calcularArea(int lados, double lado, double apotema) {
		if(lados==5) {
			double perimetro = 5* lado;
			return (perimetro * apotema)/2;
		}
		return -1;
	}
	public static void main(String[] args) {
		System.out.println(calcularArea(5)); 
        System.out.println(calcularArea(4, 6)); 
        System.out.println(calcularArea(3, 5, true)); 
        System.out.println(calcularArea(8, 5, 4, true)); 
        System.out.println(calcularArea(5, 6, 4)); 
	}
}
